### DISCUSSION TITLE
- Grott - grottproxy forward error :  [Errno 110] Connection timed out  	 - Can't establish connection with remote server.  	 - Closing connection with client side ('172.17.0.1', 34908)  	 - Grott - grottproxy forward error :  [Errno 110] Connection timed out  	 - Can't establish connection with remote server.

### DISCUSSION THREAD

**Discussion by ZonPV** [johanmeijer/grott#419] [Q&A]
This morning I get this error in the applicationlog:

```
 - Grott - grottproxy forward error :  [Errno 110] Connection timed out

	 - Can't establish connection with remote server.

	 - Closing connection with client side ('172.17.0.1', 34908)
```

Running version
	version:             	 2.7.8
	verbose:             	 True
	trace:               	 False
	config file:         	 grott.ini
	minrecl:             	 100
	decrypt:             	 True
	compat:              	 False
	invtype:             	 default
	invtypemap:          	 {}
	include_all:         	 False
	blockcmd:            	 False
	noipf:               	 False
	time:                	 auto
	sendbuf:             	 True
	timezone:            	 local
	valueoffset:         	 6
	offset:              	 6
	inverterid:          	 automatic
	mode:                	 proxy
	grottip              	 default
	grottport            	 5279
_MQTT:
	nomqtt               	 False
	mqttip:              	 192.168.1.18
        mqttport:            	 1883
	mqtttopic:           	 energy/growatt
	mqttmtopic:          	 False
	mqttmtopicname:      	 energy/meter
	mqttinverterintopic: 	 False
	mqtttretain:         	 False
	mqtttauth:           	 False
	mqttuser:            	 grott
	mqttpsw:             	 **secret**

_Growatt server:
	**growattip:           	 47.91.67.66**
	growattport:         	 5279
`
This version of the application is running in Docker on a Synology DS718 for several months without problems.

Problem: It does not send any MQTT-messages to Node-Red and does not upload data to PVoutput...

At the moment the sun is shining and on the display of the inverter it says it is running and making about 500 Watt.

Mosquitto Broker is up and passing the messages of the SMA and Omnik-inverters.

I guess there is a mistake in my configuration regarding the Growatt server? What growattip is preferred in my case?

Any help to get the application up and running again is welcome! 

Thanks in advance

**Comment by arjanhagg** [comment]
I am experiencing the same issue since approx. 14 hours ago. As the growatt server webpage [server.growatt.com](https://server.growatt.com) is not reachable, I suspect it has something to do with that issue.
What I don't understand, is why this has any effect on my local network. I would assume grott still forwards the data through mqtt even when the remote connection to server.growatt.com is not working.

Please assist, thank you!

**Reply by ZonPV** [reply]
I am not sure about this part of the Grott.ini

```
[Growatt] 
# Ip address and port of Growatt server (specify only if IP has changed)
#ip = 192.168.1.18
#port = 5279     
```                                                   
Should I remove those square hooks and restart again?

I gave it a try. But no effect. Still no MQTT-traffic and upload to PVoutput from the Growatt-application in Docker.

**Reply by arjanhagg** [reply]
I believe you should keep that part default. In my grott.ini it says:

```
[Growatt]
# Server name/IP address and port of Growatt server
# specify only if the IP address of server.growatt.com is changed
# The address as of Nov 2022 is 47.91.67.66

#ip = server.growatt.com
#ip = 47.91.67.66
#port = 5279
```

Because of your #, your config is default as well, so I suspect it does not matter what you put there as long as the line starts with a #.

**Reply by ZonPV** [reply]
That ini-file worked for years for my case. So I changed it again with square hooks.

I do not really understand the meaning of this sentence: _(specify only if IP has changed)_ 

To be honest: I am not sure if there is no data send to server.growatt.com.  Strange indeed that is seems (to me that) there still is some dependence in Grott v2.7.8  related to availability of that server in China.

**Reply by arjanhagg** [reply]
This is about the IP of the Growatt server. It's default in Grott is 47.91.67.66, but can be changed if Growatt changes it in the future for some reason.

As this Grott application is designed as a proxy, it still forwards the logs towards Growatt so you are still able to use their app. But this is the first time I see Grott failing when the Growatt server is unavailable.
I started using this app because the Growatt servers are unreliable, but now I am not sure anymore if that usecase is viable.

**Reply by mpsanten** [reply]
It seems the ip is being blocked by ISP, in my case KPN.  I also can not reach https://server.growatt.com/ from a browser

**Reply by arjanhagg** [reply]
I did a bit more troubleshooting. It seems the FQDN is not being resolved to the IP-address anymore, regardless of ISP. The server IP-address 47.91.67.66 does still answers ping packets and still listens to TCP port 5279 and 443 (https) when not coming from KPN (tested with VPN).
So it looks like there are more issues going on at the same time.

**Reply by mpsanten** [reply]
I had the resolve issues too,, but those went away after some time. Connecting directly on IP address failed via my KPN line and succeeded when I used a VPN connection.

**Reply by arjanhagg** [reply]
Resolving does seem to work again indeed. 

Still, it does not make any sense that Grott is not working properly, because it should only act locally. @johanmeijer maybe you can shine a light on what is happening here?

**Reply by arjanhagg** [reply]
I contacted the Abuse team of KPN via e-mail. I will update this thread when I have an answer.

**Update:** 
KPN says it does not block websites. It is more probable that Growatt is blocking our address. Maybe Grott is making to much calls towards the server?

**Update 2:**
Just noticed Grott started working again an hour ago. website is reachable too.

**Reply by mpsanten** [reply]
Yes,, In their forum it seems they admitted they had a problem:
https://forum.kpn.com/internet-9/growatt-is-sinds-gisteravond-niet-meer-bereikbaar-599997?postid=1826686#post1826686

**Reply by ZonPV** [reply]
Besides the possibility of blocking a site with any blacklist; 
Grott does not use resolving the name of a site. In my Grott.ini is a IP-number... 

So, in my opinion,  that cannot be the cause of this problem.

But anyway: Grott is up and running again! But still curious why we need the Growatt-servers.. 
@johanmeijer: I had in mind this was a standalone application.

**Comment by johanmeijer** [comment]
@ZonPV Grott is a proxy and  "sits" between the inverter and the growatt-servers. 

If you want to get rid of the growatt-server you might wan to use grottserver (or the grottserver docker image). For mor information see: https://github.com/johanmeijer/grott/wiki/Grottserver and discussion #98,

**Reply by mpsanten** [reply]
Thanks for that Johan,, I didn't find that yet and will look into it.

**Reply by arjanhagg** [reply]
Hi @johanmeijer, thanks for chiming in. I am aware of the grottserver and am considering using it. 

Can you maybe explain what happened yesterday? Why was Grott not forwarding the local traffic anymore? For now I am happy with the proxy setup as long as I can count on Grott to work no matter the status of the Growatt server.

**Comment by mpsanten** [comment]
The issue yesterday also learned me the importance of the IP adress in the converter. You can lock yourself out of the converter settings.
I couldn't find a topic how you can safely change that (a DNS record doesn't seem to work) Anyone here with any experience on that ?  I couldn't find a topic where this is being discussed.

**Comment by ZonPV** [comment]
> @ZonPV Grott is a proxy and "sits" between the inverter and the growatt-servers.
> 
> If you want to get rid of the growatt-server you might wan to use grottserver (or the grottserver docker image). For mor information see: https://github.com/johanmeijer/grott/wiki/Grottserver and discussion #98,

Thanks @johanmeijer 

I did not read the error PROXY. So I used your application with a misconfiguration.. 

Thanks for reminder.

**Comment by johanmeijer** [comment]
@ZonPV  Can we close this one?

**Reply by arjanhagg** [reply]
I know I am not the OP, but I would still like to know what happened on Grott side and how I could be that it stopped working presumably due to the Growatt server issues.

**Comment by johanmeijer** [comment]
@arjanhagg I am not sure what you mean with what happend at Grott site. If the growatt server is not available there is no communication between the inverter and the growatt server. 

As described Grott is a proxy and need a communication flow (picks up the data from the inverter and send it to the growatt server). 

If the growatt server does not respond the inverter will try to reconnect (you can see that in the grott log) but there will be no data flow. 

The inverter / datalogger will buffer the data and send it at the moment the communication is established again. You can see that in the Grott mqtt message: "buffered": "yes"

Growatt server can be emulated by Grottserver (separate program). If you use grottserver then you are independent from Growatt. 

Maybe this picture (from the grottserver description at the wiki) explains it better: 

![image](https://github.com/johanmeijer/grott/assets/16508298/981f6c45-29dc-41da-965f-862fba49d110)

**Reply by arjanhagg** [reply]
Thanks for the explanation. I was under the impression that Grott would always forward the inverter logs (through mqtt in my case) no matter the status of the Growatt server.

Because the Grott IP address is configured in my Shinelan box as the Growatt server, I did not expect the inverter/Shinelan tot be aware of the real Growatt server and it's reachability nor did I expect Grott to be dependent on it for it's own ability to function properly.

Now I know that not to be true, I may switch over to your Grott server setup. I don't really mind my energy info to be forwarded to the Growatt servers, but I definitely do not want to be dependent of them.
